### ISSUE TITLE
ActionFilter order over mixed sync/async filters not documented

### ISSUE THREAD

**Issue by generateui** [dotnet/AspNetCore.Docs#26559]
1. have async ActionFilter
2. have sync ActionFilter
3. apply `Order = 1` on sync, `Order = 2` on async when attributed on controller method

Note that async ActionFilter will be executed first despite that the sync has lower order (so should be first).  This specific interplay
is not documented in this article. I'm unsure what the desired behavior is (is there a bug?).

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: a70a955f-dd5a-c46b-dfe8-21521e619320
* Version Independent ID: ebb299e0-6c1c-7077-689d-ac711352e418
* Content: [Filters in ASP.NET Core](https://docs.microsoft.com/en-us/aspnet/core/mvc/controllers/filters?view=aspnetcore-6.0)
* Content Source: [aspnetcore/mvc/controllers/filters.md](https://github.com/dotnet/AspNetCore.Docs/blob/main/aspnetcore/mvc/controllers/filters.md)
* Product: **aspnet-core**
* Technology: **aspnetcore-mvc**
* GitHub Login: @Rick-Anderson
* Microsoft Alias: **riande**

**Comment by Rick-Anderson** [comment]
@generateui  can you create the simplest possible sample to demonstrate that in a public repo?

**Comment by generateui** [comment]
This failed because of an error on my side. One of the attributes I had defined did not implement `IOrderedFilter`. That was because the implementation was a private nested class which **did** implement `IOrderedFilter`. That caused me to miss it. Consequently, I implemented IOrderedFilter on the outisde exposed attribute class, and now it works as expected. 

The pattern I applied was having
```csharp
[AttributeUsage(AttributeTargets.Method)]
public class MyCustomAttribute : Attribute, IFilterFactory
{
	public IFilterMetadata CreateInstance(IServiceProvider serviceProvider)
	{
		return new Implementation();
	}
	private class Implementation : ActionFilterAttribute, IAsyncActionFilter
	{
		// implementation here
	}
}
```

A note/warning/clarification on use of `IFilterFactory` in the docs may help preventing this self-induced stupidity in the future.

Suggest to close this issue.
